static/js/components/forms/RegisterExtraDetailsForm_test.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+import React from "react"
+import sinon from "sinon"
+import { assert } from "chai"
+import { mount } from "enzyme"
+import wait from "waait"
+
+import RegisterExtraDetailsForm from "./RegisterExtraDetailsForm"
+
+import {
+  findFormikFieldByName,
+  findFormikErrorByName
+} from "../../lib/test_utils"
+
+describe("RegisterExtraDetailsForm", () => {
+  let sandbox, onSubmitStub
+
+  const renderForm = () =>
+    mount(<RegisterExtraDetailsForm onSubmit={onSubmitStub} />)
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox()
+    onSubmitStub = sandbox.stub()
+  })
+
+  it("passes onSubmit to Formik", () => {
+    const wrapper = renderForm()
+
+    assert.equal(wrapper.find("Formik").props().onSubmit, onSubmitStub)
+  })
+
+  it("renders the form", () => {
+    const wrapper = renderForm()
+
+    const form = wrapper.find("Formik")
+    assert.ok(findFormikFieldByName(form, "birth_year").exists())
+    assert.ok(findFormikFieldByName(form, "company_size").exists())
+    assert.ok(form.find("button[type='submit']").exists())
+  })
+
+  //
+  ;[
+    ["company", "", "Company is a required field"],
+    ["company", "MIT", null],
+    ["job_title", "", "Job Title is a required field"],
+    ["job_title", "QA Tester", null]
+  ].forEach(([name, value, errorMessage]) => {
+    it(`validates the field name=${name}, value=${JSON.stringify(
+      value
+    )} and expects error=${JSON.stringify(errorMessage)}`, async () => {
+      const wrapper = renderForm()
+
+      const input = wrapper.find(`input[name="${name}"]`)
+      input.simulate("change", { persist: () => {}, target: { name, value } })
+      input.simulate("blur")
+      await wait()
+      wrapper.update()
+      assert.deepEqual(
+        findFormikErrorByName(wrapper, name).text(),
+        errorMessage
+      )
+    })
+  })
+
+  //
+  ;[
+    ["gender", "", "Gender is a required field"],
+    ["gender", "f", null],
+    ["birth_year", "", "Birth Year is a required field"],
+    ["birth_year", "2000", null]
+  ].forEach(([name, value, errorMessage]) => {
+    it(`validates the field name=${name}, value=${JSON.stringify(
+      value
+    )} and expects error=${JSON.stringify(errorMessage)}`, async () => {
+      const wrapper = renderForm()
+
+      const input = wrapper.find(`select[name="${name}"]`)
+      input.simulate("change", { persist: () => {}, target: { name, value } })
+      input.simulate("blur")
+      await wait()
+      wrapper.update()
+      assert.deepEqual(
+        findFormikErrorByName(wrapper, name).text(),
+        errorMessage
+      )
```

Comment:
Take or leave: I'm wondering about this testing approach that involves rendering the full form, simulating changes for filling out the form, then checking for the validation elements. Could we instead test the formik `validationSchema` or the yup object shape directly, then test the form to make sure it includes elements with names/ids that match the schema? I understand that this may be an existing testing pattern and that it might not be quick to implement this other approach. If it seems like a heavy lift, feel free to ignore

This applies to the test case above as well

Reply:
Maybe that can be opened as a separate tech-debt issue, for now I'd prefer to keep using the same test approach used in other forms; and if the approach should be changed, change it globally.